### ISSUE TITLE
add mTLS support for client certificates

### ISSUE THREAD

**Issue by mackerel225** [element-hq/element-x-android#3463]
#### What would you like to do?
Implement mTLS for Element X by allowing users to select which client certificates to use during account provider selection screen.

#### Why would you like to do it?
Hosting matrix server will only allow people with client certificates to connect.

#### How would you like to achieve it?
HTTP-Shortcuts app has similar functionality which can be used as initial implementation - https://github.com/search?q=repo%3AWaboodoo%2FHTTP-Shortcuts%20client%20certificate%20authentication&type=code. Enables users to select which client certificate to use for HTTP requests, this enables mTLS in where your server is enforcing client certs.

#### What is the current behaviour?
Enforcing client certs on a server whilst hosting a matrix instance will result in your typical 'We couldn't reach this homeserver' error message. This will happen despite client certs are installed on an Android device, they are just not being picked up / used by Element X android.

### Are you willing to provide a PR?

Yes

**Comment by sabirovrinat85** [comment]
Implementing mTLS would be great, this is one of those features that's not much complicated (even rather simple I think) but giving so much, that one will improve overall security drastically...

**Comment by MolS64** [comment]
Yes, I also think this is a necessary feature in terms of security. If on desktop versions it works natively from under the OS, then in the case of iPhone/iPad it unfortunately does not work.

I'm trying to connect from an iPhone to my server, which is behind nginx with the ssl_verify_client directive, and I see in the logs that "client sent no required SSL certificate while reading client request headers".
At the same time, when trying to connect from a regular Element, I get a warning about the certificate in iOS, but when trying to connect via Element X, the logs are empty.

It's a shame that Element doesn't support this.

**Comment by daskladas** [comment]
The lack of client certificate authentication (mTLS) support on mobile clients is a serious security gap. mTLS is not a niche feature — it is a critical mechanism for enforcing trust and access control, and it should be available in any platform that claims to be secure and privacy-focused.

On desktop, this works seamlessly at the OS level. On mobile, however, Element and other Matrix clients completely ignore system-installed certificates, which makes it impossible to connect to a homeserver that enforces mTLS.

This limitation is a dealbreaker. Matrix has the potential to be the perfect solution, but without mTLS support on Android and iOS, it is unusable in secure environments. I strongly urge the team to prioritize this feature, as its absence undermines the credibility of Matrix as a security-first protocol.

**Comment by LibertyHomesteads** [comment]
This feature is greatly needed!

**Comment by Gunni** [comment]
I was trying to migrate my kids to an unfederated server but I don't want the service exposed...

Please add mTLS, it literally in the OS already, just ask the OS to prompt then use the response!
